code/drasil-lang/lib/Language/Drasil/Space.hs

```diff
@@ -45,6 +45,17 @@ data Space =
   | Void
   deriving (Eq, Show)
 
+-- | Sanity check to ensure we only allow numeric types inside vectNDS/ClifS
+checkClifSpace :: Space -> Bool
+checkClifSpace Real    = True
+checkClifSpace Integer = True
+checkClifSpace _       = False
+
+vectNDS :: String -> Space -> Space
+vectNDS x s
+  | checkClifSpace s = ClifS (VDim x) s
```

Comment:
I think you can use https://github.com/JacquesCarette/Drasil/blob/986746db72384c88af3cfb4f26117c125a4a25c4/code/drasil-lang/lib/Language/Drasil/Space.hs#L105-L106 to help instead of `checkClifSpace`?